exercises/dnd-character/DndCharacterTest.cs

```diff
@@ -0,0 +1,135 @@
+    public void Ability_modifier_for_score_7_is_minus_2()
+    {
+        Assert.Equal(-2, DndCharacter.Modifier(7));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_8_is_minus_1()
+    {
+        Assert.Equal(-1, DndCharacter.Modifier(8));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_9_is_minus_1()
+    {
+        Assert.Equal(-1, DndCharacter.Modifier(9));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_10_is_0()
+    {
+        Assert.Equal(0, DndCharacter.Modifier(10));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_11_is_0()
+    {
+        Assert.Equal(0, DndCharacter.Modifier(11));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_12_is_1()
+    {
+        Assert.Equal(1, DndCharacter.Modifier(12));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_13_is_1()
+    {
+        Assert.Equal(1, DndCharacter.Modifier(13));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_14_is_2()
+    {
+        Assert.Equal(2, DndCharacter.Modifier(14));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_15_is_2()
+    {
+        Assert.Equal(2, DndCharacter.Modifier(15));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_16_is_3()
+    {
+        Assert.Equal(3, DndCharacter.Modifier(16));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_17_is_3()
+    {
+        Assert.Equal(3, DndCharacter.Modifier(17));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Ability_modifier_for_score_18_is_4()
+    {
+        Assert.Equal(4, DndCharacter.Modifier(18));
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Random_ability_is_within_range()
+    {
+        for (var i = 0; i < 10; i++)
+        {
+            Assert.InRange(DndCharacter.Ability(), 3, 18);
+        }
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Random_character_is_valid()
+    {
+        for (var i = 0; i < 10; i++)
+        {
+            var sut = DndCharacter.Generate();
+            Assert.InRange(sut.Strength, 3, 18);
+            Assert.InRange(sut.Dexterity, 3, 18);
+            Assert.InRange(sut.Constitution, 3, 18);
+            Assert.InRange(sut.Intelligence, 3, 18);
+            Assert.InRange(sut.Wisdom, 3, 18);
+            Assert.InRange(sut.Charisma, 3, 18);
+            Assert.Equal(sut.Hitpoints, 10 + DndCharacter.Modifier(sut.Constitution));
+        }
+    }
+
+    [Fact(Skip = "Remove to run test")]
+    public void Each_ability_is_only_calculated_once()
+    {
+        var sut = DndCharacter.Generate();
+        Assert.Equal(sut.Strength, sut.Strength);
```

Comment:
Since this exercise is based on randomness there's actually a 1/6th chance that if it is being recalculated every time this test will pass. Does it make sense to add a few more checks to lower the probability of a false positive?

Reply:
@robkeim The PR has been updated.